gettsim/social_insurance_contributions/eink_grenzen.py

```diff
@@ -152,32 +200,165 @@ def midi_job_bemessungsentgelt_m(
-            - soz_vers_beitr_params["geringfügige_eink_grenzen_m"]["mini_job"]["west"]
+            - geringfügigkeitsgrenze_west
         )
     ) - (
-        soz_vers_beitr_params["geringfügige_eink_grenzen_m"]["mini_job"]["west"]
+        geringfügigkeitsgrenze_west
         / (
             soz_vers_beitr_params["geringfügige_eink_grenzen_m"]["midi_job"]
-            - soz_vers_beitr_params["geringfügige_eink_grenzen_m"]["mini_job"]["west"]
+            - geringfügigkeitsgrenze_west
         )
         * midi_job_faktor_f
     )
 
     return mini_job_anteil + lohn_über_mini * gewichtete_midi_job_rate
 
 
+def midi_beitragspfl_einnahme_m(
+    midi_job_faktor_f: float,
+    bruttolohn_m: float,
+    soz_vers_beitr_params: dict,
+) -> float:
+    """Total income subject to social insurance contributions for employers a
+    and employees for midi job since October 2022.
+
+    Beitragspflichtige Einnahme is the reference income for midi jobs subject
+    to employer and employee social insurance contribution.
+
+    Legal reference: Changes in § 20 SGB IV from 01.10.2022
+
+
+    Parameters
+    ----------
+    midi_job_faktor_f
+        See basic input variable :ref:`midi_job_faktor_f <midi_job_faktor_f>`.
+    bruttolohn_m
+        See basic input variable :ref:`bruttolohn_m <bruttolohn_m>`.
+    soz_vers_beitr_params
+        See params documentation :ref:`soz_vers_beitr_params <soz_vers_beitr_params>`.
+
+
+    Returns
+    -------
+    Income subject to social insurance contributions for midi job.
+    """
+    geringfüg_grenze = soz_vers_beitr_params["geringfügige_eink_grenzen_m"]["mini_job"][
+        "west"
+    ]
+
+    midijob_grenze = soz_vers_beitr_params["geringfügige_eink_grenzen_m"]["midi_job"]
+
+    quotient1 = (midijob_grenze) / (midijob_grenze - geringfüg_grenze)
+    quotient2 = (geringfüg_grenze) / (midijob_grenze - geringfüg_grenze)
+    einkommen_diff = bruttolohn_m - geringfüg_grenze
+
+    faktor1 = midi_job_faktor_f * geringfüg_grenze
+    faktor2 = (quotient1 - quotient2 * midi_job_faktor_f) * einkommen_diff
+    out = faktor1 + faktor2
+
+    return out
+
+
+def midi_sond_beitragspfl_einnahme_m(
+    bruttolohn_m: float,
+    soz_vers_beitr_params: dict,
+) -> float:
+    """Income subject to employee social insurance contributions for midi job
+    since October 2022.
+
+    Gesonderte Beitragspflichtige Einnahme is the reference income for midi jobs subject
+    to employee social insurance contribution.
+
+    Legal reference: Changes in § 20 SGB IV from 01.10.2022
+
+
+    Parameters
+    ----------
+    bruttolohn_m
+        See basic input variable :ref:`bruttolohn_m <bruttolohn_m>`.
+    soz_vers_beitr_params
+        See params documentation :ref:`soz_vers_beitr_params <soz_vers_beitr_params>`.
+
+
+    Returns
+    -------
+    Income subject to employee social insurance contributions for midi job.
+    """
+    geringfüg_grenze = soz_vers_beitr_params["geringfügige_eink_grenzen_m"]["mini_job"][
```

Comment:
Why west? Can we use `mini_job_grenze`?

Reply:
Yes, you are right, using `mini_job_grenze` is more elegant, I will change it. I just used west since it is the same for both regions in 2022, but it might change again in the future I guess, so the other solution is better